### PR TITLE
Combine check_call and print_compiler_stack. NFC.

### PR DESCRIPTION
**Pull request by sbc100**
This means that `emcc -v` will consistently print all sub-commands that
run via `check_call`.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1995,10 +1995,7 @@ def get_clang_command_asm(src_file):
         # output the dependency rule. Warning: clang and gcc behave differently
         # with -MF! (clang seems to not recognize it)
         logger.debug(('just preprocessor ' if has_dash_E else 'just dependencies: ') + ' '.join(cmd))
-        shared.print_compiler_stage(cmd)
-        rtn = run_process(cmd, check=False).returncode
-        if rtn:
-          return rtn
```
Comment (aheejin) [inline emcc.py:2001]: This previous code seems to ignore if `run_process` exists with a nonzero exit code, but the new code will call `sys.exit(1)` (inside `check_call`) in case of an error. Is this change intended?

File: emcc.py [hunk 2]
```diff
@@ -2011,8 +2008,8 @@ def get_clang_command_asm(src_file):
         if specified_target:
           cmd += ['-o', specified_target]
         logger.debug("running (for precompiled headers): " + cmd[0] + ' ' + ' '.join(cmd[1:]))
-        shared.print_compiler_stage(cmd)
-        return run_process(cmd, check=False).returncode
+        shared.check_call(cmd)
+        return 0
```
Comment (aheejin) [inline emcc.py:2012]: Here too. The old code will return the nonzero exiti code in case of an error but the new code will terminate the program.

File: tests/test_browser.py [hunk 3]
```diff
@@ -1593,7 +1593,7 @@ def test_worker(self):
     for file_data in [1, 0]:
       cmd = [EMCC, path_from_root('tests', 'hello_world_worker.cpp'), '-o', 'worker.js'] + (['--preload-file', 'file.dat'] if file_data else [])
       print(cmd)
-      subprocess.check_call(cmd)
+      self.run_process(cmd)
```
Comment (aheejin) [inline tests/test_browser.py:1596]: Why the change?

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:2001]: Returning non-zero from this function (main) is that same as exiting the whole process with non-zero.   So I'm pretty sure there is no change here, just a cleanup.

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py:2012]: Ditto, in both cases the non-zero return code is supposed be fatal and continues to be after this change.

File: tests/test_browser.py (hunk 3 above)
Comment (sbc100) [reply tests/test_browser.py:1596]: This is kind of an unrelated cleanup.. test code should always run `*self*.run_process` these days so it fails in the right way.

**Review by aheejin — APPROVED** [review]
